lib/libzrepl/mgmt_conn.c

```diff
@@ -1017,6 +1159,32 @@ process_message(uzfs_mgmt_conn_t *conn)
 		    payload_size);
 		break;
 
+	case ZVOL_OPCODE_SNAP_LIST:
+		if (payload_size == 0 || payload_size > MAX_NAME_LEN) {
+			rc = reply_nodata(conn, ZVOL_OP_STATUS_FAILED,
+			    hdrp->opcode, hdrp->io_seq);
+			break;
+		}
+		strlcpy(zvol_name, payload, payload_size);
+		zvol_name[payload_size] = '\0';
+		if ((zinfo = uzfs_zinfo_lookup(zvol_name)) == NULL) {
+			LOGERRCONN(conn, "Unknown zvol: %s", zvol_name);
+			rc = reply_nodata(conn, ZVOL_OP_STATUS_FAILED,
+			    hdrp->opcode, hdrp->io_seq);
+			break;
+		}
+		if (zinfo->mgmt_conn != conn) {
+			uzfs_zinfo_drop_refcnt(zinfo);
+			LOGERRCONN(conn, "Target used invalid connection for "
+			    "zvol %s", zvol_name);
+			rc = reply_nodata(conn, ZVOL_OP_STATUS_FAILED,
+			    hdrp->opcode, hdrp->io_seq);
+			break;
+		}
+		LOGCONN(conn, "Snaplist command for %s", zinfo->name);
+		rc = uzfs_zvol_fetch_snapshot_list(conn, hdrp, zinfo);
```

Comment:
this need to be called from async thread, as, we have a single mgmt_thread for all volumes. Doing this call here can block mgmt_thread for other volumes as well.

Reply:
please look at SNAPCREATE opcode